daemon/cmd/endpoint.go

```diff
@@ -677,8 +717,15 @@ func NewDeleteEndpointIDHandler(d *Daemon) DeleteEndpointIDHandler {
 func (h *deleteEndpointID) Handle(params DeleteEndpointIDParams) middleware.Responder {
 	log.WithField(logfields.Params, logfields.Repr(params)).Debug("DELETE /endpoint/{id} request")
 
+	r, err := h.daemon.apiLimiterSet.Wait(params.HTTPRequest.Context(), apiRequestEndpointDelete)
+	if err != nil {
+		return api.Error(http.StatusTooManyRequests, err)
+	}
+	defer r.Done()
+
 	d := h.daemon
 	if nerr, err := d.DeleteEndpoint(params.ID); err != nil {
+		r.Error(err)
```

Comment:
I don't think so, deletion will likely still have succeeded, just with some errors. The metrics currently reflects the error code so it will count errors if an error was returned which I think is correct.